File: libembosser-core-drivers/src/test/java/org/brailleblaster/libembosser/drivers/BrailleFilterInputStreamTest.java
package org.brailleblaster.libembosser.drivers;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.fail;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.Random;

import org.brailleblaster.libembosser.drivers.utils.BrailleFilterInputStream;
import org.testng.annotations.Test;

public class BrailleFilterInputStreamTest {
	@Test
	public void testReadByteConversion() {
		byte[] convertArray = new byte[256];
		for (int i = 0; i < 0x60; i++) {
			convertArray[i] = (byte)i;
		}
		for (int i = 0x60; i < 0x80; i++) {
			convertArray[i] = (byte)(i - 0x20);
		}
		for (int i = 0x80; i < 256; i++) {
			convertArray[i] = (byte)i;
		}
		byte[] input = new byte[100000];
		new Random().nextBytes(input);
		InputStream is = new BrailleFilterInputStream(new ByteArrayInputStream(input));
		int counter = 0;
		try {
			int value = is.read();
			while (value >= 0) {
				assertEquals((byte)value, convertArray[Byte.toUnsignedInt(input[counter])]);
				counter++;
				value = is.read();
			}
		} catch (IOException e) {
			fail("Unexpected exception in test", e);
		} finally {
			try {
				is.close();
			} catch (IOException e) {
				// Should never happen
			}
		}
		assertEquals(counter, input.length);
	}
}

File: libembosser-core-drivers/src/test/kotlin/org/brailleblaster/libembosser/drivers/BrailleFilterInputStreamTest.kt
package org.brailleblaster.libembosser.drivers

import org.brailleblaster.libembosser.drivers.utils.BrailleFilterInputStream
import kotlin.test.assertEquals
import kotlin.test.fail
import kotlin.test.Test
import java.io.ByteArrayInputStream
import java.io.IOException
import java.io.InputStream
import java.util.*

class BrailleFilterInputStreamTest {
    private val random = Random(System.currentTimeMillis())
    @Test
    fun testReadByteConversion() {
        val convertArray = ByteArray(256)
        for (i in 0..0x5f) {
            convertArray[i] = i.toByte()
        }
        for (i in 0x60..0x7f) {
            convertArray[i] = (i - 0x20).toByte()
        }
        for (i in 0x80..255) {
            convertArray[i] = i.toByte()
        }
        val input = ByteArray(100000)
        random.nextBytes(input)
        val `is`: InputStream = BrailleFilterInputStream(ByteArrayInputStream(input))
        var counter = 0
        try {
            var value = `is`.read()
            while (value >= 0) {
                assertEquals(value.toByte(), convertArray[java.lang.Byte.toUnsignedInt(input[counter])])
                counter++
                value = `is`.read()
            }
        } catch (e: IOException) {
            fail("Unexpected exception in test", e)
        } finally {
            try {
                `is`.close()
            } catch (e: IOException) {
                // Should never happen
            }
        }
        assertEquals(counter, input.size)
    }
}